util/mathutil/math.go

```diff
@@ -90,10 +90,10 @@ func Min[v constraints.Ordered](x v, xs ...v) v {
 }
 
 // Clamp restrict a value to a certain interval.
-func Clamp[v constraints.Integer | constraints.Float](n, min, max v) v {
-	if n >= max {
+func Clamp[T constraints.Ordered](n, min, max T) T {
+	if n > max {
```

Comment:
Now you bring more comparison if n == max or n == min. not important but we shouldn't make code worse.

Reply:
if `n == max` or `n == min`, then `return max or min` is the same as `return n`, which is the default return.
so they are equal.
I thouhgt `>=` may be faster than `>`, but after benchmark, the are same too.
i prefer keep `>` and `<` since it is more similiar with max and min

Reply:
thanks. you explain is reasonable, restored to original compare logic.